pkg/label/label.go

```diff
@@ -13,3 +13,29 @@ const (
 	// ManagedBy is set for Kubernetes resources managed by the operator.
 	ManagedBy = "giantswarm.io/managed-by"
 )
+
+// ProcessLabels ensures the chart-operator.giantswarm.io/version label is
+// present and the app-operator.giantswarm.io/version label is removed. It
+// also ensures the giantswarm.io/managed-by label is accurate.
+//
+// Any other labels added to the app custom resource are passed on to the chart
+// custom resource.
+func ProcessLabels(projectName, chartCustomResourceVersion string, inputLabels map[string]string) map[string]string {
```

Comment:
```suggestion
func ProcessLabels(inputLabels, requiredLabels, excludedLabels map[string]string) map[string]string {
```
This is cool but I'd like to see it made more generic so we can reuse in all resources. WDYT to using this method signature?

For example for the chart CRs we need to replace the app-operator version label with the chart-operator version. But we don't need to do that for the configmaps.

We should also have unit tests for this.

Reply:
Good idea.